### PR TITLE
Upgrade ES to 5.1.1

### PR DESCRIPTION
**Pull request by marcinc**
This PR:
* Bumps ES to latest 5.1.1
* Removes elasticsearch-HQ as ES 5.x doesn't run site plugins. Possible to add that as standalone later.
* Changes ES / logging configuration
* `ES_HEAP_SIZE` deprecated, now `ELASTIC_SEARCH_HEAP_SIZE`. Heap size set via `ES_JAVA_OPTS`

- [x] Depends on: https://github.com/UKHomeOffice/aws-dsp/pull/200

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -6,11 +6,10 @@ RUN adduser -d /data -m elasticsearch
 
 EXPOSE 9200 9300
 
-ENV ES_VERSION 2.4.1
-RUN wget -q https://download.elasticsearch.org/elasticsearch/release/org/elasticsearch/distribution/tar/elasticsearch/${ES_VERSION}/elasticsearch-${ES_VERSION}.tar.gz -O - | tar -xzf -; mv elasticsearch-${ES_VERSION} /elasticsearch && mkdir -p /elasticsearch/config/scripts /elasticsearch/plugins
-RUN /elasticsearch/bin/plugin install https://oss.sonatype.org/service/local/repositories/releases/content/io/fabric8/elasticsearch-cloud-kubernetes/2.4.1/elasticsearch-cloud-kubernetes-2.4.1.zip
-RUN /elasticsearch/bin/plugin install com.floragunn/search-guard-ssl/2.4.1.16 && (cd /elasticsearch/plugins/search-guard-ssl/ && wget -q http://repo1.maven.org/maven2/io/netty/netty-tcnative/1.1.33.Fork17/netty-tcnative-1.1.33.Fork17-linux-x86_64-fedora.jar)
-RUN /elasticsearch/bin/plugin install royrusso/elasticsearch-HQ/v2.0.3
```
Comment (jits) [inline Dockerfile:13]: Could replace this with [kopf](https://github.com/lmenezes/elasticsearch-kopf)?

Comment (jits) [reply Dockerfile:13]: Although, doesn't look like kopf supports the latest Elastic stack.

File: README.md [hunk 2]
```diff
@@ -88,3 +88,18 @@ $ kubectl create -f kube/es-svc.yaml
 $ kubectl create -f kube/es-client-rc.yaml
 $ kubectl create -f kube/es-data-rc.yaml
 ```
+
+### Elasticsearch Index configuration
```
Comment (vaijab) [inline README.md]: Not really needed, because this is documented in ES docs and irrelevant here.

File: config/log4j2.properties [hunk 3]
```diff
@@ -0,0 +1,82 @@
+status = error
```
Comment (vaijab) [inline config/log4j2.properties:1]: Did you make any changes to this file or is the same that ES ships already?

What we want is less verbose logging and to console only.

File: run.sh [hunk 4]
```diff
@@ -4,7 +4,8 @@ export CLUSTER_NAME=${CLUSTER_NAME:=elasticsearch}
 export NODE_NAME=${NODE_NAME:=${HOSTNAME}}
 export NODE_LOCAL=${NODE_LOCAL:=false}
 export PATH_DATA=${PATH_DATA:=/data}
-export ES_HEAP_SIZE=${ES_HEAP_SIZE:=450m}
+export ELASTIC_SEARCH_HEAP_SIZE=${ELASTIC_SEARCH_HEAP_SIZE:=450m}
```
Comment (vaijab) [inline run.sh:7]: What's the point in renaming the env var?

File: run.sh [hunk 5]
```diff
@@ -24,4 +25,6 @@ export DISCOVERY_ZEN_PUBLISH_TIMEOUT=${DISCOVERY_ZEN_PUBLISH_TIMEOUT:=30s}
 export ENABLE_TRANSPORT_SSL=${ENABLE_TRANSPORT_SSL:=false}
 
 chown -R elasticsearch:elasticsearch /data
+chmod +r /elasticsearch/config/log4j2.properties /elasticsearch/config/elasticsearch.yml
```
Comment (vaijab) [inline run.sh]: What's that?

**Review by vaijab — CHANGES_REQUESTED** [review]

File: run.sh (hunk 4 above)
Comment (marcinc) [reply run.sh:7]: `ES_HEAP_SIZE` is no longer supported and ES complaints about it as heap size is now set via java opts. Renamed to keep it separately configurable.

File: README.md (hunk 2 above)
Comment (marcinc) [reply README.md]: Will remove.

File: run.sh (hunk 4 above)
Comment (vaijab) [reply run.sh:7]: Ah cool. This time they check that the world is correct :-)

File: config/log4j2.properties (hunk 3 above)
Comment (marcinc) [reply config/log4j2.properties:1]: It's new default logging configuration shipped with ES 5.x. I'll have a look how to reduce logging here.

Comment (vaijab) [reply config/log4j2.properties:1]: If we don't change anything then we should not add this file.

File: run.sh (hunk 5 above)
Comment (marcinc) [reply run.sh]: Early testing artifact... Removing.

File: config/log4j2.properties (hunk 3 above)
Comment (marcinc) [reply config/log4j2.properties:1]: It's been slightly modified before but now reduced it to only log to console. Now using official docker  image configuration (https://github.com/docker-library/elasticsearch/blob/1107caed5c403a63d99c1b1ae1f7002d387c7e85/5/config/log4j2.properties).

**Review by vaijab — APPROVED** [review]

**Pull request merged**